entity-service-impl/src/main/java/org/hypertrace/entity/data/service/EntityDataServiceImpl.java

```diff
@@ -711,28 +784,30 @@ private <T extends Message> void searchByIdAndStreamSingleResponse(
       String entityType,
       Collection collection,
       Message.Builder builder,
-      StreamObserver<T> responseObserver) {
+      StreamObserver<T> responseObserver)
+      throws IOException {
```

Comment:
rather than introducing a throw here and lots of new catches, since this is already responsible for sending the response to the observer (not ideal, but existing), can we just send the exception to the observer in here too. Do any catches of this method differ?